src/kmerseek/entity.py

```diff
@@ -0,0 +1,42 @@
+import os
+
+import polars as pl
+
+from .sketch import sketch, _make_sigfile
+from .sig2kmer import get_kmers, _make_kmer_filename
+
+
+class KmerseekEntity:
+    """Base class to be inherited by KmerseekQuery and KmerseekIndex"""
+
+    def __init__(self, fasta, moltype, ksize, scaled):
+        # These are all filenames of where the data is stored
+        self.fasta = fasta
+        self.sketch_kws = dict(moltype=moltype, ksize=ksize, scaled=scaled)
+
+    @property
+    def sig(self):
+        """String of k-mer signature filename"""
+        if not hasattr(self, "_sig"):
+            self._sig = sketch(self.fasta, **self.sketch_kws)
```

Comment:
Of course, this works, but would it be preferable to use `@sig.setter` instead? [docs that make me think this](https://docs.python.org/3/library/functions.html#property)